WIP on user feedback (#1021)
mongodb/docs-realm: source/sdk/dotnet/fundamentals/object-models-and-schemas.txt

<dotnet-object-schema>` that defines the properties and relationships of those
objects. In C#, you define object schemas by using the native class implementation.

When {+service-short+} is initialized, it discovers the {+service-short+} objects 
defined in all assemblies that have been loaded.

.. note:: 

   .NET does not load an assembly until you reference a class in it, so if you 

>    define your object models in one assembly and instantiate Realm in another, 

Reviewer: ```suggestion
   define your object models in one assembly and instantiate {+service-short+} in another, 
```
Source constants!

   be sure to call a method in the assembly that contains the object models

>    *before* initialization. Otherwise, Realm will not discover the objects 

Reviewer: ```suggestion
   *before* initialization. Otherwise, {+service-short+} will not discover the objects 
```
Source constants!

>    *before* initializing Realm. Otherwise, Realm will not discover the objects 

Reviewer: ```suggestion
   *before* initializing {+service-short+}. Otherwise, {+service-short+ will not discover the objects 
```

Reviewer: Note: Perhaps there are more usages of the term "Realm" here than we need.

Author: Perhaps....but I'm blocked mentally. Suggestions?

Author: OK, I removed one of them. :)

>    *before* initializing Realm. Otherwise, Realm will not discover the objects 

Reviewer 2: Do you need to actually call a method from the other assembly before you initialize Realm? Or is just referencing them enough?

Also my .NET is rusty - is a reference here a `using` statement or something else?

Author: @nlarew just saw the second part of your Q. Referencing is adding the assembly to your current assembly. If it's an external assembly, like the Realm SDK, you use NuGet. Otherwise, you can just add another project in the solution as a dependency.

- ``DateTimeOffset``
- ``char``
- ``string``

> - ``Guid``

Reviewer: Is this datatype available yet? I suspect this is meant for the new data types branch.

Author: Yoips! Good catch!

- ``ObjectId``

.. seealso::

   :ref:`Usage Examples: Index a Property <dotnet-index-property-example>`

.. _dotnet-decimal128:

Using Decimal Values
--------------------
Realm supports 128-bit decimal values with the ``Decimal128`` Bson type. When 
defining a decimal type, you can use the ``Decimal128`` Bson type or the .NET 
``decimal`` type, even though it is only a 96-bit decimal. The SDK automatically 
converts between the two, although you risk losing precision or range. The 
following example shows how to use both the ``Decimal128`` Bson type and the .NET 
``decimal`` type:

.. literalinclude:: /examples/generated/code/start/DecimalFun.codeblock.decimal128.cs
   :language: csharp

> 

Reviewer: What happened to decimal128 values!?!?!

Author: They were moved to the document where they belonged a long time ago, but never removed from this doc. Just some campsite cleanup. :)
